state.go

```diff
@@ -569,7 +575,7 @@ func (a *AuthorizedState) Handle(ctx *StateContext, c Controller) (State, bool)
 	// handle easy case first, no update info present, means no update in progress
 	if err != nil && os.IsNotExist(err) {
 		log.Debug("no update in progress, proceed")
-		return updateCheckWaitState, false
+		return inventoryUpdateState, false
```

Comment:
State transitions go like this:
* `init -> bootstrap -> authorized --(no update)--> inventory update -> update check wait`
* and after a complete update cycle
`init -> boostrap -> authorized -- (with update) --> <apply, commit, etc.> (clear update) -> update status report -> init` (see above)
* or with errors
`init -> boostrap -> authorized -- <some update states> -> error state (clear update) -> init` (see above)

I believe I asked for input from you and @kacf when starting this task, the recommendation was to add another state into existing SM. I'm also not sure if it's ok to send inventory reports while applying an update. We would need extra synchronization to prevent this from happening if inventory was to run in parallel.